custom_components/foxess_em/__init__.py

```diff
@@ -77,36 +79,45 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     # Check FoxESS API Key before settings up the other platforms to prevent those being setup if it
     # fails, which will prevent setup working when the FoxESS API key is fixed.
-    if len(entry.options) > 0:
+    entry_options = copy.deepcopy(dict(entry.options))
+    entry_data = copy.deepcopy(dict(entry.data))
+    _LOGGER.debug(f"Options {entry.options}")
+    _LOGGER.debug(f"Data {entry.data}")
+
+    if entry_options != entry_data:
```

Comment:
Thanks Dave - it's been a while since I python'd but will this compare dict values or the actual objects? i.e. will it always evaluate to false?

Reply:
lol, even then your python would still be considerably better than mine ;)

I think the answer to that can be quite complex dependent on the content, but for this case it is easier - as long as both dict values contain the same keys in the same order (which they do) it does do a deep compare.

so

options = {'a':1, 2:2, 3:'help'}
data = {'a':1, 2:2, 3:'help'}
options==data # is true
options!=data # is false

options = {'a':1, 2:2, 3:'help'}
data = {'a':1, 2:2, 3:'nohelp'}
options==data # is false
options!=data # is true

Reply:
actually hold that thought, this would work on a running system but I need to check what happens when first set up

Reply:
Hmm ok, when you first set up the integration entry.options is an empty dict so that fails the compare - only when you have run the config again and it has been updated would the `!=` work.

It needs another check to make sure that entry.options is not empty to cover that initial state - as an empty dict evaluates False in python this works correctly.

```
    if entry_options and entry_options != entry_data:
        # overwrite data with options
        entry_data = copy.deepcopy(dict(entry.options))
        _LOGGER.info("Config has been updated")
    _LOGGER.debug("Data %s", entry_data)

```

I've tested that ok for initial and changed states - am I missing any other else ?

Reply:
@nathanmarlor Hi Nath, could you set this as a pre-release please and i'll test run it for a couple of days.